ERCS/erc-7844.md

```diff
@@ -0,0 +1,529 @@
+	* Custom getter/setter functions, which increase deployment costs while requiring integrators to implement protocol-specific logic to retrieve state.  
+
+	More fundamentally, access patterns may be subject to change—an integration built for one contract version may break when the underlying storage structure is updated, requiring maintenance across multiple versions.
+
+4. **Manual Storage Management** 
+	Storage updates are inherently high-risk due to Solidity’s strict alignment rules. Even when following best practices, upgradeable storage models require careful management to avoid introducing permanent corruption.
+
+	While Proxy-Delegate and Diamond Storage architectures provide upgrade paths, they require explicit tracking of storage layouts. Misaligned struct modifications or incorrect slot allocations can cause irrecoverable state corruption.
+
+	Additionally, the need for manual slot assignments and upgrade procedures introduces a security risk—any miscalculation in struct offsets, slot positions, or initialization routines can lead to unintended storage overwrites. While careful audits and best practices reduce this risk, they do not eliminate it entirely.
+
+	It is true that best practices can largely mitigate the risks associated with storage updates. However, the potential for breaking changes remains a technical burden in itself—ensuring upgrade safety requires significant time, effort, and expertise, particularly in complex ecosystems.
+
+5.  **Siloed Storage**
+	On-chain storage is often tied to individual contracts rather than a unified data management layer.  This leads to complex linked structures and multi-hop access patterns, where operations require data aggregated from multiple external dependencies.  This added complexity and computational overhead is purely a burden, and can be avoided by leveraging dedicated storage layers with a cross-contract scope, as is often seen in the Diamond Pattern.  
+
+6.  **Long-Term Technical Debt and Maintainability** 
+	Factors including adapters, siloed storage, multi-hops, and dependent interfaces introduce new risks and considerations, increasing the complexity of upgrades as ecosystem scope expands.  The absence of these considerations streamlines the upgrade process considerably, allowing developers to prioritize optimizing and securing critical business logic.  Additionally, increased lines of code and complexity typically garner higher audit costs, while increasing the odds that vulnerable code makes it to production.
+
+
+## Specification 
+
+![Adaptability](../assets/erc-7844/Adaptability.svg)
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY” and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Extendable Structs and Storage Spaces
+- Extendable structs **MUST** leverage dynamic mappings with deterministic field hashes (`keccak256`).  
```

Comment:
Don't bold the uppercase keywords. Just leave them uppercase, but otherwise unadorned.